Analysis/OverlayRemoval/src/CheatedMCOverlayRemoval.cc

```diff
@@ -0,0 +1,344 @@
+  m_nRun++ ;  
+  streamlog_out(DEBUG0) << "   processRunHeader finished successfully" << std::endl ;
+}
+
+void CheatedMCOverlayRemoval::processEvent( LCEvent *pLCEvent )
+{
+  this->Clear();
+  m_nRun = pLCEvent->getRunNumber();
+  m_nEvt = pLCEvent->getEventNumber();
+  streamlog_out(DEBUG) << "Processing event " << pLCEvent->getEventNumber() << std::endl;
+
+  const EVENT::LCCollection *PFOs{};
+  const EVENT::LCCollection *MCPs{};
+  IMPL::LCCollectionVec* OutputPfoCollection(NULL);
+  OutputPfoCollection = new IMPL::LCCollectionVec(LCIO::RECONSTRUCTEDPARTICLE);
+  OutputPfoCollection->setSubset( true ); 
+  IMPL::LCCollectionVec* OutputOverlayCollection(NULL);
+  OutputOverlayCollection = new IMPL::LCCollectionVec(LCIO::RECONSTRUCTEDPARTICLE);
+  OutputOverlayCollection->setSubset( true ); 
+  try
+    {
+      PFOs = pLCEvent->getCollection( _recoParticleCollectionName );
+      MCPs = pLCEvent->getCollection( _MCParticleCollectionName );
+
+      LCRelationNavigator RecoMCParticleNav( pLCEvent->getCollection( _recoMCTruthLink ) );
+      LCRelationNavigator MCParticleRecoNav( pLCEvent->getCollection( _mcTruthRecoLink ) );
+
+      m_nAllPFOs = PFOs->getNumberOfElements();
+      m_nMCPs = MCPs->getNumberOfElements();
+      int nKeptPFOs = 0;
+      int nRemovedPFOs = 0;
+      
+      //Create new collection without PFOs from MCPs that are overlay
+      for (int i=0; i<m_nMCPs; i++) {
+	MCParticle* mcp = (MCParticle*) MCPs->getElementAt(i);
+	if (mcp->isOverlay()) { 
+	  float weightPFOtoMCP = 0.0;
+	  float weightMCPtoPFO = 0.0;
+	  ReconstructedParticle* linkedPFO = getLinkedPFO(mcp, RecoMCParticleNav, MCParticleRecoNav, weightPFOtoMCP, weightMCPtoPFO);
+	  if ( linkedPFO == NULL ) {
+	    continue;
+	  }
+	  streamlog_out(DEBUG5) << "i = " << i << ": energy = " << linkedPFO->getEnergy () << std::endl;
+	  OutputOverlayCollection->addElement(linkedPFO);
+	  nRemovedPFOs++;
+	}
+      }
+
+      for (int i=0; i<m_nAllPFOs; i++) {
+	ReconstructedParticle* pfo = (ReconstructedParticle*) PFOs->getElementAt(i);
+	float weightPFOtoMCP = 0.0;
+	float weightMCPtoPFO = 0.0;
+	MCParticle* linkedMCP = getLinkedMCP(pfo, RecoMCParticleNav, MCParticleRecoNav, weightPFOtoMCP, weightMCPtoPFO);
+	if ( linkedMCP != NULL ) {
+	  if (linkedMCP->isOverlay()) {
+	    streamlog_out(DEBUG5) << "i = " << i << ": energy = " << pfo->getEnergy () << ", gen status = " << linkedMCP->getGeneratorStatus() << std::endl;
+	    continue;
+	  }
+	OutputPfoCollection->addElement(pfo);
+	nKeptPFOs++;
+      }
+
+      streamlog_out(DEBUG5) << "In event " << m_nEvt << ": Kept PFOs = " << nKeptPFOs << " VS removed number of PFOs = " << nRemovedPFOs << " VS total number of PFOs = " << m_nAllPFOs << std::endl;
+      pLCEvent->addCollection(OutputPfoCollection, _OutputPfoCollection.c_str() );
+      pLCEvent->addCollection(OutputOverlayCollection, _OutputOverlayCollection.c_str() );
+      m_nEvt++ ;
+    }
+  catch(lcio::DataNotAvailableException& e)
+    {
```

Comment:
There seems to be a '}' missing here or potentially a `;` somewhere in the `try` block, because the compiler is complaining here:
https://github.com/iLCSoft/MarlinReco/actions/runs/4881127837/jobs/8712928598#step:4:361

could you check and try to fix this?

Reply:
Fixed the missing `}` but am having issues with MarlinUtil to compile it myself...